### ISSUE TITLE
[New Sensor]: linp.magnet.m1

### ISSUE THREAD

**Issue by louliangsheng** [custom-components/ble_monitor#1006]
### Sensor Description

Linptech Door and Window Sensor

### Additional information

https://home.miot-spec.com/spec?type=urn:miot-spec-v2:device:magnet-sensor:0000A016:linp-m1:1

[home-assistant.log](https://github.com/custom-components/ble_monitor/files/9484068/home-assistant.log)

![微信图片_20220904125657](https://user-images.githubusercontent.com/37586254/188298246-b9afe925-215f-4ebd-88b8-80ac56c62f7b.jpg)
![微信图片_20220904125646](https://user-images.githubusercontent.com/37586254/188298247-03a44ce5-765b-4981-82d7-108b085d23ed.jpg)

### BLE advertisements
[home-assistant.log](https://github.com/custom-components/ble_monitor/files/9484068/home-assistant.log)

**Comment by Ernst79** [comment]
I checked the data and it seems we can add this sensor. 

Do you have a link to a product specs, I can't seem to find anything. 
It needs an encryption key BTW

**Comment by Ernst79** [comment]
I think it is called Linptech M1, or LINP-M1. 

Can you test it with 10.0.1-beta (enable beta versions in HACS to in install beta versions). I assumed that it is using the "opening" object. You will need to set the encryption key to use it, see the FAQ on how to get the encryption key. 

Let me know if it works, than I can merge it into a final version.

**Comment by louliangsheng** [comment]
I updated the bete version but it is not recognized correctly, I added the key and the key is correct.
The log I uploaded before is the log I got after adding the key. I don't know if there is a problem. If there is any impact, I will re-upload the log without the key.
In addition, this door and window sensor has a wireless button and supports clicking. I don't know if it is not correct because of this.
Hope what I said helps you.
Also here is the link where I purchased this door and window sensor.
https://m.xiaomiyoupin.com/detail?gid=156202&spmref=YouPin_I.share.share_pop_copy.4.22919278

![image](https://user-images.githubusercontent.com/37586254/188416601-a21e1798-596a-4c2c-b634-90f3d02e3d9c.png)
![image](https://user-images.githubusercontent.com/37586254/188416651-3fcdd197-d920-41d0-9f47-8b7865e59042.png)

**Comment by Ernst79** [comment]
Thanks, I have enough info to decrypt the data. Ill look into it

**Comment by Ernst79** [comment]
I talked to @rezmus, and he ran it through his miot script

```
prop.2.1004 (Status) uint8 = eid 18436 (0x4804)
event.2.1018 (Open) = eid 18962 (0x4a12)
event.2.1015 (Device Forcibly Removed) = eid 18959 (0x4a0f)
event.2.1019 (Close) = eid 18963 (0x4a13)
event.2.1026 (Door Not Closed) = eid 18970 (0x4a1a)
prop.3.1003 (Battery Level) uint8 = eid 19459 (0x4c03)
```

So, we have 6 different types of data.

obj id 4804 = eid 18436 = status
obj id 4a0f = eid 18959 = Device Forcibly Removed
obj id 4a12 = eid 18962 = open
obj id 4a13= eid 18963 = closed
obj id 4a1a = eid 18970 = Door Not Closed
obj id 4c03 = eid 19459 = Battery Level

**Comment by Ernst79** [comment]
I've created a new beta release 10.0.2-beta. Can you check if this one works (it should). 

One important thing is that I used both obj 4808 and 4a12 for the opening binary sensor, 4a13 is most likely the button, so I assigned that to a "button sensor". The Door not closed and device forcibly removed (I called it "door/window broken open") should be an attribute of the opening sensor (attribute is called status). 

Can you check if everything works as expected?

**Comment by louliangsheng** [comment]
I updated the beta version of the plugin and the device is working fine, thank you for your work

**Comment by Ernst79** [comment]
Released as 10.1.0 as final. 

@louliangsheng Did you also check the button?

**Comment by louliangsheng** [comment]
Yes.it works well.

Ernst Klamer ***@***.***>于2022年9月6日 周二18:57写道：
